gradescopeCronJob/gradescope_to_spreadsheet.py

```diff
@@ -228,6 +229,71 @@ def make_batch_request(sheet_api_instance):
     make_request(batch_request)
     logger.info(f"Completing batch request")
 
+def push_all_grade_data_to_sheets():
+    gradescope_client = initialize_gs_client()
+    assignment_id_to_names = get_assignment_id_to_names(gradescope_client)
+    sheet_api_instance = create_sheet_api_instance()
+    get_sub_sheet_titles_to_ids(sheet_api_instance) #
+    populate_spreadsheet_gradebook(assignment_id_to_names)
+    make_batch_request(sheet_api_instance) #
+
+    # An assignment is marked as current if there are >3 submissions and if the commented code in the for loop below is removed
+    assignment_id_to_currency_status = {}
+    for id in assignment_id_to_names:
+        prepare_request_for_one_assignment(sheet_api_instance, gradescope_client=gradescope_client,
+                                                               assignment_name=assignment_id_to_names[id], assignment_id=id)
+    make_batch_request(sheet_api_instance)
+
+
+def populate_spreadsheet_gradebook(assignment_id_to_names):
+
+    # The below code is used to filter assignments by category when populating the instructor dashboard
+    assignment_names = assignment_id_to_names.values()
+    filter_by_assignment_category = lambda category: lambda assignment: category in assignment.lower()
+
+    labs = set(filter(filter_by_assignment_category("lab"), assignment_names))
+    projects = set(filter(filter_by_assignment_category("project"), assignment_names))
+    lecture_quizzes = set(filter(filter_by_assignment_category("lecture"), assignment_names))
+    midterms = set(filter(filter_by_assignment_category("midterm"), assignment_names))
+
+    def extract_number_from_assignment_title(assignment):
+        numbers_present = re.findall("\d+", assignment)
+        if numbers_present:
+            return int(numbers_present[0])
+        return 0
+
+    sorted_labs = sorted(labs, key=extract_number_from_assignment_title)
+    sorted_projects = sorted(projects, key=extract_number_from_assignment_title)
+    sorted_lecture_quizzes = sorted(lecture_quizzes, key=extract_number_from_assignment_title)
+    sorted_midterms = sorted(midterms, key=extract_number_from_assignment_title)
+
+    formula_list = [GRADE_RETRIEVAL_SPREADSHEET_FORMULA] * NUMBER_OF_STUDENTS
+
+    def produce_gradebook_for_category(sorted_assignment_list, category):
+        global subsheet_titles_to_ids
+        grade_dict = {name : formula_list for name in sorted_assignment_list}
+        grade_df = pd.DataFrame(grade_dict).set_index(sorted_assignment_list[0])
+        output = io.StringIO()
+        grade_df.to_csv(output)
+        grades_as_csv = output.getvalue()
+        output.close()
+        assemble_rest_request_for_assignment(grades_as_csv, sheet_api_instance=None, sheet_id=subsheet_titles_to_ids[category], rowIndex=0, columnIndex=3)
```

Comment:
suggestion: include a logging statement such as f"Successfully processed {category} on the instructor dashboard". The logging message in `assemble_rest_request_for_assignment` is not detailed enough to help with debugging since there are 4 different categories we are processing.